File: 01362.cpp
#include <iostream>
using namespace std;

int main() {
	int scenario = 1;
	while (1) {
		int o, w; cin >> o >> w;
		if (!o && !w) break;

		bool dead = false;
		while (1) {
			char c; int n;
			cin >> c >> n;
			if (c == '#' && !n) break;

			if (c == 'E') w -= n;
			else w += n;

			if (w <= 0) dead = true;
		}

		cout << scenario++ << ' ';
		if (dead) cout << "RIP\n";
		else if (w > o / 2 && w < o * 2) cout << ":-)\n";
		else cout << ":-(\n";
	}
}

File: 01362.py
scenario = 1
while 1:
    o, w = map(int, input().split())
    if not o and not w: break

    dead = False
    while 1:
        c, n = input().split()
        n = int(n)
        if c == '#' and not n: break

        if c == 'E': w -= n
        else: w += n

        if w <= 0: dead = True

    print(scenario, end=' ')
    scenario += 1

    if dead: print("RIP")
    elif w > o//2 and w < o*2: print(":-)")
    else: print(":-(")
